frontend/amundsen_application/static/js/pages/TableDetailPage/LineageButton/index.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+
+import { TableMetadata } from 'interfaces';
+import { isTableLineagePageEnabled } from 'config/config-utils';
+import { logClick } from 'utils/analytics';
+
+export interface LineageButtonProps {
+  tableData: TableMetadata;
+}
+
+const LineageButton: React.FC<LineageButtonProps> = ({
+  tableData,
+}: LineageButtonProps) => {
+  if (!isTableLineagePageEnabled()) return null;
+
+  const { database, cluster, schema, name } = tableData;
+  const href = `/lineage/table/${cluster}/${database}/${schema}/${name}`;
+  if (!href) return null;
+
+  const label = 'Lineage';
+  return (
+    <a
+      className="btn btn-default btn-lg"
+      href={href}
+      role="button"
+      id="table-lineage-button"
+      onClick={logClick}
```

Comment:
We are trying to move out of using logClick as the only tracker, using Redux actions and their analytics payload to track.

See an example on https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/ducks/bookmark/reducer.ts#L28

You will need to pass down the dispatcher down to this component and probably create an action for it. See (https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/ducks/lineage/reducer.ts#L67)

That said, I am wondering why we are following an href here instead of using a router link and avoiding a full reload.